### PR TITLE
[Framework] Port 8572 ocean startup when saas

### PR DESCRIPTION
**Pull request by omby8888**
# Description

What - handle startup when running as saas

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: port_ocean/core/utils.py [hunk 1]
```diff
@@ -29,6 +30,17 @@ def is_same_entity(first_entity: Entity, second_entity: Entity) -> bool:
     )
 
 
+async def validate_integration_runtime(
+    port_client: PortClient, requested_runtime: Runtime
+):
+    current_integration = await port_client.get_current_integration()
+    current_runtime = current_integration.get("installationType", "OnPrem")
+    if current_integration and current_runtime != requested_runtime:
```
Comment (yairsimantov20) [inline port_ocean/core/utils.py:38]: Why wont u have the integration

File: port_ocean/core/utils.py [hunk 2]
```diff
@@ -29,6 +30,17 @@ def is_same_entity(first_entity: Entity, second_entity: Entity) -> bool:
...
+        raise Exception(
+            f"Can't run {current_runtime} integration in {requested_runtime} runtime"
```
Comment (yairsimantov20) [inline port_ocean/core/utils.py:40]: Please use custom exception that inherit ocean exceptions

File: port_ocean/run.py [hunk 3]
```diff
@@ -45,6 +47,11 @@ def run(
         "app", default_app
     )
 
+    # Validate that the current integration's runtime matches the execution parameters
+    asyncio.get_event_loop().run_until_complete(
+        validate_integration_runtime(app.port_client, app.config.runtime)
+    )
```
Comment (yairsimantov20) [inline port_ocean/run.py:53]: It means the application already filled in the defaults 
You need to catch that in the integration initialization

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "port-ocean"
-version = "0.7.1"
+version = "0.7.2"
```
Comment (yairsimantov20) [inline pyproject.toml:3]: Its a new feature therefore it should be a new minor

File: port_ocean/core/utils.py (hunk 1 above)
Comment (omby8888) [reply port_ocean/core/utils.py:38]: it doesn't necessarily exist, what if this is the first time running...

File: port_ocean/core/utils.py [hunk 5]
```diff
@@ -29,6 +30,17 @@ def is_same_entity(first_entity: Entity, second_entity: Entity) -> bool:
     )
 
 
+async def validate_integration_runtime(
+    port_client: PortClient, requested_runtime: Runtime
+) -> None:
+    current_integration = await port_client.get_current_integration(should_raise=False)
+    current_runtime = current_integration.get("installationType", "OnPrem")
+    if current_integration and current_runtime != requested_runtime:
+        raise OceanAbortException(
```
Comment (yairsimantov20) [inline port_ocean/core/utils.py:39]: ocean abort exception are for resyncs
can you create a new dedicated exception derived from BaseOceanException?

**Review by yairsimantov20 — APPROVED** [review]
small comment
and update the changelog date

File: port_ocean/core/utils.py [hunk 6]
```diff
@@ -29,6 +30,17 @@ def is_same_entity(first_entity: Entity, second_entity: Entity) -> bool:
     )
 
 
+async def validate_integration_runtime(
+    port_client: PortClient, requested_runtime: Runtime
+) -> None:
```
Comment (yairsimantov20) [inline port_ocean/core/utils.py:35]: maybe add a debug log

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,13 @@ this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 <!-- towncrier release notes start -->
 
+## 0.9.0 (2024-06-18)
```
Comment (yairsimantov20) [inline CHANGELOG.md:10]: update the date

File: port_ocean/core/utils.py [hunk 8]
```diff
@@ -29,6 +30,17 @@ def is_same_entity(first_entity: Entity, second_entity: Entity) -> bool:
...
+    current_integration = await port_client.get_current_integration(should_raise=False)
+    current_runtime = current_integration.get("installationType", "OnPrem")
+    if current_integration and current_runtime != requested_runtime:
+        raise OceanAbortException(
+            f"Can't run {requested_runtime} integration in {current_runtime} runtime"
```
Comment (yairsimantov20) [inline port_ocean/core/utils.py:40]: Be more precise and verbose

**Pull request merged**